controllers/openstackdataplanerole_controller.go

```diff
@@ -216,21 +219,72 @@ func (r *OpenStackDataPlaneRoleReconciler) Reconcile(ctx context.Context, req ct
 			return result, nil
 		}
 
-		r.Log.Info("Set DataPlaneRoleReadyCondition true")
-		instance.Status.Conditions.Set(condition.FalseCondition(dataplanev1beta1.DataPlaneRoleReadyCondition, condition.InitReason, condition.SeverityInfo, dataplanev1beta1.DataPlaneRoleReadyWaitingMessage))
-	}
+		// Call DeployNovaExternalCompute individually for each node
+		var novaExternalCompute *novav1beta1.NovaExternalCompute
+		var novaReadyConditionsTrue []*condition.Condition
+		var novaErrors []error
+		for _, node := range nodes.Items {
+			nodeConfigMapName := fmt.Sprintf("dataplanenode-%s-inventory", node.Name)
+			result, novaExternalCompute, err = deployment.DeployNovaExternalCompute(
+				ctx,
+				helper,
+				&node,
+				ansibleSSHPrivateKeySecret,
+				nodeConfigMapName,
+				&instance.Status,
+				instance.Spec.NetworkAttachments,
+				instance.Spec.OpenStackAnsibleEERunnerImage)
+			if err != nil {
+				novaErrors = append(novaErrors, err)
+			}
+			novaReadyCondition := novaExternalCompute.Status.Conditions.Get(condition.ReadyCondition)
+			if novaReadyCondition != nil && novaReadyCondition.Status == corev1.ConditionTrue {
+				r.Log.Info(fmt.Sprintf("NovaExternalCompute %s, ReadyCondition status %s", node.Name, novaReadyCondition.Status))
+				novaReadyConditionsTrue = append(novaReadyConditionsTrue, novaReadyCondition)
+
+			}
+		}
+
+		// When any errors are found, wrap all into a single error, and return
+		// it
+		errStr := "DeployNovaExternalCompute error:"
+		if len(novaErrors) > 0 {
+			for _, err := range novaErrors {
+				errStr = fmt.Sprintf("%s: %s", errStr, err.Error())
```

Comment:
I believe the final error message would be something like this:
`DeployNovaExternalCompute error: stacktrace1: stacktrace2:...:stacktraceN`

Reply:
I think that's ok. Not sure how else to represent it.